polynote-frontend/polynote/cell.js

```diff
@@ -335,6 +339,27 @@ export class CodeCell extends Cell {
             this.editor.updateOptions({ renderLineHighlight: "none" });
         });
 
+        this.editor.onDidChangeCursorSelection(evt => {
+            // we only care if the user has selected more than a single character
+            if ([0, 3].includes(evt.reason) && // 0 -> NotSet, 3 -> Explicit.
+                (evt.selection.startLineNumber !== evt.selection.endLineNumber || evt.selection.startColumn !== evt.selection.endColumn)) {
+
+                console.log(evt)
+                let hash = document.location.hash;
+                if (!hash) {
+                    hash = `${this.container.id}`
+                } else {
+                    hash = hash.split(",")[0];
+                }
+                if (evt.selection.startLineNumber === evt.selection.endLineNumber) {
+                    document.location.hash = `${hash},${evt.selection.startLineNumber}`
```

Comment:
Doesn't this really mess up the history? Probably need to use the history API... `history.replaceState` or something.

Reply:
Indeed, each time you select a cell or text you'll get a new entry in the history. Let me see what I can do with replaceState

Reply:
Fixed!